Create space item model setup

AnyaAven opened the pull request:
Create the base setup for space-to-item relationship; one to many.

This involved factories, seeders, migrations, and models

AnyaAven committed:
Space and Item model setup WIP

Need to add foreign key to an Item model

AnyaAven committed:
Update dependencies

AnyaAven committed:
Seed script

AnyaAven committed:
Foreign Key added space -> items

AnyaAven committed:
Merge branch 'main' of https://github.com/AnyaAven/item-nook into create-space-item-model-setup

# Conflicts:
#	composer.lock

AnyaAven committed:
misc

AnyaAven committed:
misc

Riley19280 commented on database/factories/ItemFactory.php:
```diff
@@ -0,0 +1,47 @@
+            'description' => fake()->sentence(),
+            'notes'       => fake()->paragraph(),
+            'image_path'  => fake()->image(),
+            'custom_data' => json_encode($customData),
```
I don't think that this needs to be json encoded if the json cast is set up on the model

AnyaAven replied:
I see! Will do.

AnyaAven committed:
Fix json cast

The factory can store custom_data column as an array and Laravel will handle the interaction of the array while storing it as json.
